### ISSUE TITLE
Abnormal CPU consumption and potential deadlock on RDB load 

### ISSUE THREAD

**Issue by filipecosta90** [RedisGraph/RedisGraph#1186]
After bulk loading a large graph with the following properties into RedisGraph 66fd60fa7aef8e25d4d92df6170cdf48ef4f3260:

Name | Description and Source | Vertices | Edges
-- | -- | -- | --
k-hop | Synthetic Kronecker graph http://graph500.org | 2.4 M | 64 M

And persisting the graph key and loading it into an older version (98043a5174e713d2d6e4452958482b51aa0037eb) I'm observing high CPU usage and a potential deadlock. The following chart illustrates on the left a standalone redis-server abnormally consuming 48VCPUs, and on the right ( after stopping the previous redis-server process ) an normal redis-server doing the RDB load. 
The following sections describe with more detail where are we spending the CPU cycles during the "abnormal behaviour"

### higher level overview of cpu cycles
(`perf top --sort comm,dso -d 5 -F 999 --pid $(pgrep redis-server)`)
```
Samples: 1M of event 'cycles', 999 Hz, Event count (approx.): 3107481182626 lost: 0/0 drop: 0/0
Overhead  Command          Shared Object
  47.28%  redis-server     libgomp.so.1.0.0
  40.72%  redis-server     redisgraph.so
   8.43%  redis-server     [kernel]
   3.56%  redis-server     libc-2.27.so
   0.01%  redis-server     redis-server
   0.01%  jemalloc_bg_thd  [kernel]
   0.00%  redis-server     libpthread-2.27.so
   0.00%  redis-server     libm-2.27.so
   0.00%  redis-server     bpf_prog_3a335fb93bbc46df
   0.00%  jemalloc_bg_thd  redis-server
```

### deep overview of cpu cycles
(`perf top -d 5 -F 999 --pid $(pgrep redis-server)`)
```
Samples: 1M of event 'cycles', 999 Hz, Event count (approx.): 3912516294590 lost: 0/0 drop: 0/0
Overhead  Shared Object              Symbol
  26.87%  libgomp.so.1.0.0           [.] 0x0000000000018f22
  12.75%  redisgraph.so              [.] GB_ewise_slice._omp_fn.0
  11.44%  redisgraph.so              [.] GB_add_phase1._omp_fn.0
  10.30%  libgomp.so.1.0.0           [.] 0x00000000000190ca
   4.97%  redisgraph.so              [.] GB_cumsum._omp_fn.0
   4.54%  redisgraph.so              [.] GB_AaddB__second_uint64._omp_fn.8
   3.63%  libgomp.so.1.0.0           [.] 0x0000000000018f26
   3.07%  redisgraph.so              [.] GB_cumsum._omp_fn.1
   2.69%  libc-2.27.so               [.] 0x000000000018ef2d
   2.24%  redisgraph.so              [.] GB_add_phase0._omp_fn.2
   2.15%  libgomp.so.1.0.0           [.] 0x0000000000018f2d
   1.61%  libgomp.so.1.0.0           [.] 0x0000000000018f2b
   1.40%  libgomp.so.1.0.0           [.] 0x00000000000190ce
   1.11%  libgomp.so.1.0.0           [.] 0x00000000000190d6
   0.86%  [kernel]                   [k] native_queued_spin_lock_slowpath
   0.73%  [kernel]                   [k] clear_page_erms
   0.72%  libgomp.so.1.0.0           [.] 0x0000000000018f20
   0.66%  redisgraph.so              [.] GB_hyper_prune._omp_fn.0
   0.63%  libgomp.so.1.0.0           [.] 0x00000000000190d3
   0.51%  [kernel]                   [k] prepare_exit_to_usermode
   0.50%  [kernel]                   [k] down_read_trylock
   0.49%  [kernel]                   [k] async_page_fault
```

### Reproduce on standalone OSS Redis and RedisGraph with the proper RDB
```
wget https://s3.amazonaws.com/benchmarks.redislabs/redisgraph/datasets/k-hop/graph500_s22_dump.rdb 
mv graph500_s22_dump.rdb dump.rdb
redis-server --loadmodule <redisgraph.so>
```

**Comment by swilly22** [comment]
The reason for the high CPU usage is the introduction of transposed matrices by #877 specifically [this line](https://github.com/RedisGraph/RedisGraph/pull/877/files?file-filters%5B%5D=.c#diff-38470088df61fe1ce66cefc99a3eab28R572)

This was fixed shortly after by #1148, as such please use either commits following b5fdffad0e8091b483fe8d27524d99705aedff68 or commits prior to 121786aa55519c98bfd339c35b4595eff70b9368

**Comment by filipecosta90** [comment]
> The reason for the high CPU usage is the introduction of transposed matrices by #877 specifically [this line](https://github.com/RedisGraph/RedisGraph/pull/877/files?file-filters%5B%5D=.c#diff-38470088df61fe1ce66cefc99a3eab28R572)
> 
> This was fixed shortly after by #1148, as such please use either commits following [b5fdffa](https://github.com/RedisGraph/RedisGraph/commit/b5fdffad0e8091b483fe8d27524d99705aedff68) or commits prior to [121786a](https://github.com/RedisGraph/RedisGraph/commit/121786aa55519c98bfd339c35b4595eff70b9368)

confirmed that your recommendations fixed it @swilly22 . closing the ticket
